workshop/Lab3/README.md

```diff
@@ -0,0 +1,252 @@
+# Lab 3: Scale and update apps natively, building multi-tier applications.
+
+In this lab, understand how to build applications using basic building blocks which are Kubernetes objects such as pods, deployments and services. These objects are defined in configuration files using yaml. Once defined, these objects are created using Kubernetes CLI. In this lab we will explore how to build a multi-tier application that can scale.
+
+Before we work with the application we need to clone a github repo from which we can get all our configuration files.
+
+``` 
+	git clone git@github.com:IBM/guestbook.git
+```
+
+# 1. Scale apps natively
+
+Kubernetes accomplishes scaling out through a deployment.  By specifying a  replica number higher than 1 to run that many pods, you can ensure your deployment has the available resources to handle increasing load on your application.
+
+1. Consider the following deployment configuration for guestbook application
+    ```
+    apiVersion: apps/v1
+	kind: Deployment
+	metadata:
+	  name: guestbook
+	  labels:
+	    app: guestbook
+	spec:
+	  replicas: 3
+	  selector:
+	    matchLabels:
+	      app: guestbook
+	  template:
+	    metadata:
+	      labels:
+	        app: guestbook
+	    spec:
+	      containers:
+	      - name: guestbook
+	        image: ibmcom/guestbook:v1
+	        ports:
+	        - name: http-server
+	          containerPort: 3000
+    ```
+The above configuration file create a deployment object named 'guestbook' with a pod containing a single container running the image 'ibmcom/guestbook:v1'.  Also the configuration specifies replicas set to 3 and kubernetes tries to make sure that at least three active pods running at any time.
```

Comment:
Since this is the first time we've introduced the config files, I think it would be good to talk about their
general structure a bit. Just something quick about how they have metadata, a spec and when you GET it you'll see a status section.  Then talk about the other stuff, but work outside-in. Start with what they should already (sort of) know, the definition of the container and then work your way out explaining what's going on and why.

Reply:
it might also be good, someplace, to mention that even in the previous exercises a Deployment was created and they can see it if they do: kubectl get deployments